### PR TITLE
Prep A/B test for collections rail, use Jest/ts

### PR DESCRIPTION
**Pull request by eessex**
- Sets up `EDITORIAL_COLLECTIONS_RAIL` in `running_tests.coffee`
- Sets up `splitTest.view()` in `src/desktop/apps/article/routes.js`

Points above are commented out for now, and need uncommented to turn the test on.

- Passes test outcome to "standard", "feature" and "news" articles as `showCollectionsRail` boolean
- Converts touched files to `tsx`
- Converts related tests to `.jest.tsx`

### CODE REVIEW

File: src/desktop/apps/article/routes.js [hunk 1]
```diff
@@ -124,12 +126,18 @@ export async function index(req, res, next) {
...
+      EDITORIAL_COLLECTIONS_RAIL, // TODO: Remove after A/B test
+      IS_MOBILE,
+      IS_TABLET,
+    } = res.locals.sd
 
     const isMobile = IS_MOBILE
     const isTablet = IS_TABLET
     const jsonLD = stringifyJSONForWeb(articleModel.toJSONLD())
     const customEditorial = isCustomEditorial(article.id)
+    const showCollectionsRail = Boolean(EDITORIAL_COLLECTIONS_RAIL)
```
Comment (xtina-starr) [inline src/desktop/apps/article/routes.js]: I wonder if it's worth combining the logic I have in the `CanvasFooter` to ensure this only displays for standard, feature, and news layouts here.

```
const showCollectionsRail = ["standard", "feature", "news"].includes(article.layout)
```

Comment (xtina-starr) [reply src/desktop/apps/article/routes.js]: Actually we don't have access to the layout at this point, correct?

Comment (eessex) [reply src/desktop/apps/article/routes.js]: This is a great idea, and yes, we do have access to the layout -- will add this

File: src/desktop/apps/article/components/InfiniteScrollArticle.tsx [hunk 2]
```diff
@@ -97,7 +112,7 @@ export default class InfiniteScrollArticle extends React.Component {
         return (
           <Waypoint
             onEnter={this.fetchNextArticles}
-            threshold={2.0}
+            // threshold={2.0}
```
Comment (xtina-starr) [inline src/desktop/apps/article/components/InfiniteScrollArticle.tsx]: Any particular reason why this is commented out?

Comment (eessex) [reply src/desktop/apps/article/components/InfiniteScrollArticle.tsx]: I commented this out while working on it -- after converting to typescript it looks like the Waypoint library no longer accepts this property.  Tested it and the UI seems to act as expected without it.

**Pull request merged**